server/queries/db.js

```diff
@@ -17,23 +18,94 @@ sequelize.sync({ alter: true }).then(() => {
     console.log("Database connected.");
 });
 
-module.exports.createUser = username => {
+module.exports.validatePassword = (username, password) => {
     return new Promise((resolve, reject) => {
-        User.findOrCreate({
+        User.findOne({
             where: {
                 username: username
-            },
+            }
+        })
+        .then(user => {
+            if (user) {
+                bcrypt.compare(password, user.dataValues.password, (err, res) => {                    
+                    if (res) {
+                        resolve(user);
+                    } else {
+                        reject({
+                            code: 401,
+                            message: "Password is incorrect"
+                        });
+                    }
+                });
+            } else {
+                reject({
+                    code: 400,
+                    message: "User with this username does not exist"
+                });
+            }
+        })
+        .catch(err => {
+            reject({
+                code: 400,
+                message: err.message
+            });
+        });
+    });
+}
+
+module.exports.createUser = (username, password) => {
+    return new Promise((resolve, reject) => {
+        bcrypt.hash(password, 10, (err, hash) => {
+            if (err) {
+                reject({
+                    code: 400,
+                    message: err.message
+                });
+            }
+            User.findOrCreate({
+                where: { username },
+                defaults: { // set the default properties if it doesn't exist
+                    username,
+                    password: hash,
+                    score: 0
+                }
+            })
+            .then(([user, created]) => {                
+                if (!created) {
+                    reject({
```

Comment:
if you `return reject()` then you don't need the else, because flow ends inside the conditional block